libraries/AP_BattMonitor/AP_BattMonitor_DJIBattery.cpp

```diff
@@ -0,0 +1,96 @@
+#include <AP_HAL/AP_HAL.h>
+#include "AP_BattMonitor_DJIBattery.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <AP_Math/crc.h>
+
+/*
+  "battery" monitor for DJI Battery via serial port.
+
+   For future Referance:
+   Total Voltage : 8,9
+   Percentage : 17
+   Cell1 :  20,21
+   Cell2 :  22,23
+   Cell3 :  24,25
+   Cell4 :  26,27
+   
+ */
+extern const AP_HAL::HAL& hal;
+
+/// Constructor
+AP_BattMonitor_DJIBattery::AP_BattMonitor_DJIBattery(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state, AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
+{
+    const AP_SerialManager &serial_manager = AP::serialmanager();
+    
+    if ((port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_DJIBattery, 0))) {
```

Comment:
Maybe use the `BUS` parameter to say which serial instance is being selected? That way multiple DJI batteries can be used?

Reply:
yes, I would like to add this in future update

Reply:
Should just be `_params._i2c_bus` to use it now.